features/license_check.feature

```diff
@@ -0,0 +1,84 @@
+
+    @wip
+    @series.lts
+    @uses.config.machine_type.gcp.generic
+    Scenario Outline: license_check job should run periodically on gcp generic lts
+        Given a `<release>` machine with ubuntu-advantage-tools installed
+        # verify its enabled
+        Then I verify the `ua-license-check` systemd timer is scheduled to run within `10` minutes
+        # run it and verify that it didn't disable itself
+        When I run `systemctl start ua-license-check.service` with sudo
+        Then I verify the `ua-license-check` systemd timer is scheduled to run within `10` minutes
+        # verify attach disables it
+        When I attach `contract_token` with sudo
+        When I run `sleep 1` with sudo
+        Then I verify the `ua-license-check` systemd timer is disabled
+        # verify detach reenables it
+        When I run `ua detach --assume-yes` with sudo
+        When I run `sleep 1` with sudo
+        Then I verify the `ua-license-check` systemd timer is scheduled to run within `10` minutes
+        # verify deleting marker file and stopping disables it
+        When I run `rm /var/lib/ubuntu-advantage/marker-license-check` with sudo
+        When I run `systemctl stop ua-license-check.timer` with sudo
+        When I run `sleep 1` with sudo
+        Then I verify the `ua-license-check` systemd timer is disabled
+        # verify creating marker file enables it
+        When I run `touch /var/lib/ubuntu-advantage/marker-license-check` with sudo
+        When I run `sleep 1` with sudo
+        Then I verify the `ua-license-check` systemd timer is scheduled to run within `10` minutes
+        Examples: version
+            | release |
+            | xenial  |
+            | bionic  |
+            | focal   |
+
+    @wip
+    @series.all
+    @uses.config.machine_type.lxd.container
+    @uses.config.machine_type.lxd.vm
+    @uses.config.machine_type.aws.generic
+    @uses.config.machine_type.azure.generic
+    Scenario Outline: license_check is disabled everywhere but gcp generic
+        Given a `<release>` machine with ubuntu-advantage-tools installed
+        Then I verify the `ua-license-check` systemd timer is disabled
+        # verify creating marker file enables it, but it disables itself
+        When I run `touch /var/lib/ubuntu-advantage/marker-license-check` with sudo
+        Then I verify the `ua-license-check` systemd timer is scheduled to run within `10` minutes
+        When I run `systemctl start ua-license-check.service` with sudo
+        When I run `sleep 1` with sudo
+        Then I verify the `ua-license-check` systemd timer is disabled
+        Examples: version
+            | release |
+            | xenial  |
+            | bionic  |
+            | focal   |
+            | hirsute |
+
+    @wip
+    @series.lts
+    @uses.config.machine_type.aws.pro
+    @uses.config.machine_type.azure.pro
+    @uses.config.machine_type.gcp.pro
+    Scenario Outline: license_check is disabled everywhere but gcp generic
```

Comment:
The pro tests require the config setup and ua auto-attach steps to get into the state that we want to test: after auto attach, the timer is disabled.

Reply:
I was thinking that maybe we could add the config setup to all machines and instead of running `auto-attach` on the PRO types, just running a normal attach. But you are right, it is better to exercise this other path